code/_onclick/hud/radial.dm

```diff
@@ -309,4 +309,45 @@ GLOBAL_LIST_EMPTY(radial_menus)
 	GLOB.radial_menus -= uniqueid
 	return answer
 
+/**
+ * Similar to show_radial_menu, but choices is a list of atoms, for which icons will be automatically generated.
+ * Supports multiple items of the same name, 2 soaps will become soap (1) and soap (2) to the user
+ */
+/proc/radial_menu_helper(mob/user, atom/anchor, list/choices, uniqueid, radius, datum/callback/custom_check, require_near = FALSE)
+	var/list/duplicate_amount = list()
+	for(var/atom/atom in choices)
+		if(!duplicate_amount.Find(atom.name))
+			duplicate_amount[atom.name] = 0
+		else
+			duplicate_amount[atom.name]++
+	var/list/duplicate_indexes = duplicate_amount.Copy()
+
+	var/list/icon_state_choices = list()
+	var/list/return_choices = list()
+	for(var/atom/atom in choices)
```

Comment:
I know it's not really a reserved keyword, but can you make use of a different variable name that isn't just `atom`?